private/engine/config.go

```diff
@@ -1,52 +1,121 @@
 package engine
 
 import (
+	"fmt"
 	"os"
 	"path/filepath"
+	"strings"
 
 	"github.com/BurntSushi/toml"
 )
 
+type connectionType uint
+
+const (
+	socketConnection connectionType = iota
+	httpConnection
+)
+
 type Config struct {
-	Socket  string `toml:"socket"`  // socket filename
-	WorkDir string `toml:"workdir"` // path to socket file
+	ConnectionType connectionType
+	SocketConfig   socketConfig `toml:",omitempty"`
+	HttpConfig     httpConfig   `toml:",omitempty"`
+}
 
-	DialTimeout           uint // timeout for connecting to socket (seconds)
-	RequestTimeout        uint // timeout for writing to socket (seconds)
-	ResponseHeaderTimeout uint // timeout for reading from socket (seconds)
+type socketConfig struct {
```

Comment:
I intentionally used separate sections in the toml as a number of fields are exclusive to either the socket, or to the http config. If I don't use separate sections in the config then it means:
1. A load of validation code would need to be added, to ensure that config fields have only been defined for the relevant type of connection.
2. It would be more confusing and potentially lead to more user errors.

I actually feel the config file is easier to set up this way, and cleaner. It's certainly simpler to document.